appinventor/appengine/src/com/google/appinventor/client/Ode.java

```diff
@@ -1445,10 +1376,22 @@ public static boolean getUserAutoloadProject() {
    *
    * @param enable true if autoloading should be enabled or false if it should be disabled.
    */
-  public static void setUserAutoloadProject(boolean enable) {
+  public static void setUserAutoloadProject(final boolean enable) {
     userSettings.getSettings(SettingsConstants.USER_GENERAL_SETTINGS)
         .changePropertyValue(SettingsConstants.USER_AUTOLOAD_PROJECT, Boolean.toString(enable));
-    userSettings.saveSettings(null);
+    userSettings.saveSettings(new Command() {
+      @Override
+      public void execute() {
+        DropDownButton settings = Ode.getInstance().getTopToolbar().settingsDropDown;
+        if (enable) {
+          settings.setCommandById("AutoloadLastProject", new DisableAutoloadAction());
+          settings.setItemHtmlById("AutoloadLastProject", MESSAGES.disableAutoload());
+        } else {
+          settings.setCommandById("AutoloadLastProject", new EnableAutoloadAction());
+          settings.setItemHtmlById("AutoloadLastProject", MESSAGES.enableAutoload());
+        }
+      }
+    });
```

Comment:
Hmm. I'm not sure about this change. Previously, the menu managed its own state but this moves the management of state to Ode, which I don't think is appropriate.

Reply:
> Hmm. I'm not sure about this change. Previously, the menu managed its own state but this moves the management of state to Ode, which I don't think is appropriate.

It seems like this should be handled in the action class itself, so I did that. It doesn't look to me like there should be two commands here, just one that toggles the setting.

I think I did this because I just end up grabbing classes from Ode anyway, but hopefully as more refactoring happens, that won't be the case.